lib/services/livesync/android-livesync-library.ts

```diff
@@ -0,0 +1,396 @@
+				.on("data", (chunk: string | Buffer) => {
+					fileHash.update(chunk);
+					if (this.socketConnection) {
+						this.socketConnection.write(chunk);
+					} else {
+						const error = this.checkConnectionStatus();
+						//TODO Destroy method added in node 8.0.0.
+						//when we depricate node 6.x uncoment the line belolw
+						//fileStream.destroy(error);
+						reject(error);
+					}
+				})
+				.on("end", () => {
+					if (this.socketConnection) {
+						this.socketConnection.write(fileHash.digest(), () => {
+							resolve(true);
+						});
+					} else {
+						const error = this.checkConnectionStatus();
+						reject(error);
+					}
+				})
+				.on("error", (error: Error) => {
+					reject(error);
+				});
+		});
+	}
+
+	private createSocket(port: number): IDuplexSocket {
+		const socket = new net.Socket();
+		socket.connect(port, this.configuration.localHostAddress);
+		return socket;
+	}
+
+	private checkConnectionStatus() {
+		if (this.socketConnection === null) {
+			const defaultError = this.getErrorWithMessage("No socket connection available.");
+			const error = this.socketError || defaultError;
+
+			return error;
+		}
+	}
+
+	private verifyActiveConnection(rejectHandler?: any) {
+		const error = this.checkConnectionStatus();
+		if (error && rejectHandler) {
+			rejectHandler(error);
+		}
+
+		if (error && !rejectHandler) {
+			this.$errors.failWithoutHelp(error.toString());
+		}
+	}
+
+	private handleConnection({ socket, data }: { socket: IDuplexSocket, data: NodeBuffer | string }) {
+		this.socketConnection = socket;
+		this.socketConnection.uid = this.generateOperationIdentifier();
+
+		const versionLength = (<NodeBuffer>data).readUInt8(0);
+		const versionBuffer = data.slice(PROTOCOL_VERSION_LENGTH_SIZE, versionLength + PROTOCOL_VERSION_LENGTH_SIZE);
+		const appIdentifierBuffer = data.slice(versionLength + PROTOCOL_VERSION_LENGTH_SIZE, data.length);
+
+		const protocolVersion = versionBuffer.toString();
+		const appIdentifier = appIdentifierBuffer.toString();
+		this.$logger.trace(`Handle socket connection for app identifier: ${appIdentifier} with protocol version: ${protocolVersion}.`);
+
+		this.socketConnection.on("data", (connectionData: NodeBuffer) => this.handleData(socket.uid, connectionData));
+		this.socketConnection.on("close", (hasError: boolean) => this.handleSocketClose(socket.uid, hasError));
+		this.socketConnection.on("error", (err: Error) => {
+			const error = new Error(`Socket Error:\n${err}`);
+			if (this.configuration.errorHandler) {
+				this.configuration.errorHandler(error);
+			} else {
+				this.handleSocketError(socket.uid, error.message);
+			}
+		});
+	}
+
+	private connectEventuallyUntilTimeout(factory: () => IDuplexSocket, timeout: number): Promise<{socket: IDuplexSocket, data: NodeBuffer | string}> {
+		return new Promise((resolve, reject) => {
+			let lastKnownError: Error | string,
+				isResolved = false;
+
+			setTimeout(() => {
+				if (!isResolved) {
+					isResolved = true;
```

Comment:
`isResolved` -> `isComplete` in order to avoid`isResolved = true` when rejected?